File: lib/graphics.lua
local ffi = require 'ffi'
local C = ffi.C

ffi.cdef[[
typedef struct _ev_graphics ev_graphics;

ev_graphics* ev_graphics_create(void);
void ev_graphics_destroy(ev_graphics*);
void ev_graphics_clear(ev_graphics*, float, float, float, float);
]]

local _M = {}
_M.__index = _M

function _M:clear(r, g, b, a)
   C.ev_graphics_clear(self._ev_graphics, r, g, b, a)
end

function _M.create()
   local g = {}
   setmetatable(g, _M)

   g._ev_graphics = C.ev_graphics_create()

   return g
end

return _M

File: src/graphics.c
#include "graphics.h"
#include <string.h>

struct _ev_graphics {
#ifdef WIN32
    int _unused;
#endif
};

ev_graphics* ev_graphics_create(void)
{
    ev_graphics *g = ev_malloc(sizeof(ev_graphics));

    if( g ) {
        memset(g, 0, sizeof(ev_graphics));
    }
    return g;
}

void ev_graphics_destroy(ev_graphics *g)
{
    ev_free(g);
}
void ev_graphics_clear(ev_graphics *graphics, float r, float g, float b, float a)
{
    UNUSED(graphics);

    glClearColor(r, g, b, a);
    glClear( GL_COLOR_BUFFER_BIT );
}
